packages/next/src/build/index.ts

```diff
@@ -368,6 +368,42 @@ function getCacheDir(distDir: string): string {
   return cacheDir
 }
 
+function getNumberOfWorkers(config: NextConfigComplete) {
+  if (
+    config.experimental.cpus &&
+    config.experimental.cpus !== defaultConfig.experimental!.cpus
+  ) {
+    return config.experimental.cpus
+  }
+
+  if (config.experimental.memoryBasedWorkersCount) {
+    return Math.max(
+      Math.min(config.experimental.cpus || 1, Math.floor(os.freemem() / 1e9)),
+      // enforce a minimum of 4 workers
+      4
+    )
+  }
+
+  if (config.experimental.cpus) {
+    return config.experimental.cpus
+  }
+
+  // Fall back to 4 workers if a count is not specified
+  return 4
+}
+
+async function writeFileUtf8(filePath: string, content: string) {
+  await fs.writeFile(filePath, content, 'utf-8')
+}
+
+function readFileUtf8(filePath: string) {
+  return fs.readFile(filePath, 'utf8')
+}
+
+async function readManifest<T extends object>(filePath: string): Promise<T> {
```

Comment:
I think we have some method for this that handles caching

Reply:
Seems we have this one: https://github.com/vercel/next.js/blob/100c9327d0d5190fb92a17fb75e1c6f203276653/packages/next/src/server/load-manifest.ts#L5 but that one is more geared towards having the long-running process and the memory is never cleaned up because it uses a module-scope `Map`, I don't think we need to cache these right now 👍